### PR TITLE
Fetch all PR timeline event

### PR DESCRIPTION
**Pull request by manzan46**
## Context

For now it was only fetching first 100 events entry for the PR

## Description

- FIxing that with `octokit.paginate` 
- Add a simple test on stats

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
 const core = require('@actions/core');
 const github = require('@actions/github');
 
-const octokit = new github.getOctokit(core.getInput('token'));
+let octokit;
+if (process.env.NODE_ENV != 'test') {
+  octokit = new github.getOctokit(core.getInput('token'));
```
Comment (manzan46) [inline src/api.js:6]: There might be a better way to that 🙈

File: src/api.js [hunk 2]
```diff
@@ -15,11 +18,14 @@ module.exports = {
   },
 
   getPullRequestEventsTimeline: async pr_number => {
-    const { data } = await octokit.rest.issues.listEventsForTimeline({
-      ...github.context.repo,
-      issue_number: pr_number,
-      per_page: 100, // TODO: fetch all.
-    });
+    const data = await octokit.paginate(
+      octokit.rest.issues.listEventsForTimeline,
```
Comment (manzan46) [inline src/api.js:22]: Main change of the PR :)

File: src/stats.js [hunk 3]
```diff
@@ -14,43 +14,11 @@ const findFirstInTimeline = ({ timeline, fn }) => {
...
-        })
-    );
-
-    return result;
-  },
-
-  getPullRequestStats: async pr => {
-    const prNumber = pr.number;
+module.exports = {
+  getStatsFromEventsTimeline: (pr, timeline) => {
```
Comment (manzan46) [inline src/stats.js:18]: Nothing change really in this file , I mainly extracted the stats calculation to be able to add a simple test

**Comment by tchapeaux** [comment]
Hello @manzan46 , I'm not too familiar with Github Actions and the Github API, so it's a bit hard for me to review this. Is there a way for me to test / debug the code?

**Comment by manzan46** [comment]
👋🏻  @tchapeaux  I didn't find an easy good way to test unfortunately, when it was on my repo I was just creating PR and merging them to test, might be the only good solution, create a private repo `test` and setup the github action with the following: 

```
name: Pull Request Stats

on:
  pull_request:
    types: [closed]

jobs:
  stats:
    runs-on: ubuntu-latest
    steps:
      - name: Run pull request stats
        uses: wooclap/pr-stats@BRANCH_NAME 
        with:
          comment: true
```

For the review itself you can ignore the `dist/index.js` it's bassically compile code by  `ncc`, there is also a bit of prettier that was left over. 

Also happy to a first pair review if it can help

**Review by tchapeaux — APPROVED** [review]
As discussed, OK for me to merge!

**Pull request merged**